File: JAVA/Search/binarySearch.java
package JAVA.Search;
public class binarySearch {
    public static boolean binary_search(int arr[],int key){
        int start = 0;
        int end = arr.length - 1;

        while(start <= end){
            int mid = (start + end) / 2;
            if(arr[mid] == key){
                return true;
            }
            else if (arr[mid] > key){
                end = mid -1;
            }
            else if(arr[mid] < key){
                start = mid+1;
            }
        }
        return false;
    }
    public static void main(String[] args){
        int a[] = {10,20,30,40,50,60,70,80};
        System.out.println(binary_search(a,90));
    }
}

File: 100 Questions C++/Binary_search.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std;


int binarySearch(int arr[],int n,int key){
    int start = 0,end=n-1;
    int mid;
    
    while(start<=end){
        mid = (start+end)/2;
        if(key>arr[mid]){
            start = mid+1;
        }else if(key<arr[mid]){
            end=mid-1;
        }else {
            return mid;
        }
    }
    return -1;

}

int main(){
    int arr[] = {5,12,23,34,56,87,90};
    int n = sizeof(arr)/sizeof(arr[0]);

    int key;
    cin>>key;
    int index = binarySearch(arr,n,key);
    if(index!=-1){
        cout<<arr[index];
    }else{
        cout<<"nout found";
    }
}